map/main_test.ts

```diff
@@ -0,0 +1,24 @@
+import { assertObjectMatch } from "https://deno.land/std@0.183.0/testing/asserts.ts";
+import { decode_map } from "../build/dev/javascript/gleam_stdlib/gleam_stdlib.mjs";
+
+Deno.test("does not throw when decoding null", () => {
+  const result = decode_map(null);
+
+  assertObjectMatch(result, {
+    0: {
+      head: { expected: "Map", found: "Null", path: {} },
+      tail: {},
+    },
+  });
+});
+
+Deno.test("does not throw when decoding undefined", () => {
+  const result = decode_map(undefined);
+
+  assertObjectMatch(result, {
+    0: {
+      head: { expected: "Map", found: "Nil", path: {} },
+      tail: {},
+    },
+  });
+});
```

Comment:
Remove these please, testing should be done in Gleam.

Reply:
The thing is I can't. Since there is no way to produce `null` I have to write the tests in JS/TS.

The implementation I had before was buggy, these tests were the only thing showing me that they were. 

I'm writing a very simple app and stumbled on multiple issues in the stdlib's JS implementation while doing this, I don't feel comfortable removing the tests here, because it could hurt gleam's early user experience if regressions are introduced.

How about I'm just removing the `undefined` test, because that one is covered by the gleam test I wrote. The `null` one I'm afraid has to stay or some parts of the JS code are not tested and a regression could be introduced at any time again.

Would that work?

Reply:
I removed the redundant one and kept the one that can't be written in gleam: 393126e7cbc7a7a9591b581a5dfe5bec8aed602d

Reply:
Alternatively I could also add an external function that returns `null` which would only be used for testing, e.g. not exported.

Would be better?

It would mean I'd need some help with Erlang though, as I'd also need a function in erlang that returns `atom`.

I assumed adding tests in TS would be OK, since there were added just recently for property testing.

Reply:
Yup, you got it. FFI can be used to construct `null` using a function in `gleam_stdlib_test_ffi.mjs`.

Creating an atom is a bit easier. Custom types without fields are atoms, so one can be made like so:

```gleam
pub type MyAtom {
  ThisIsAnAtom
}

pub fn atom_decode_test() {
  let atom = dynamic.from(ThisIsAnAtom)
  todo
}
```

Reply:
Thanks for the example. I've implemented the test. I also remove the deno tests completely.

Let me know what you think :)